From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ROCKNIX <rocknix@rocknix.org>
Date: Tue, 04 Mar 2026 00:00:00 +0000
Subject: [PATCH] arm64: dts: rockchip: rk3566: enable DMC devfreq on Anbernic
 devices

Enable the DMC (Dynamic Memory Controller) devfreq driver on all
Anbernic RK3566 handhelds (RG353P/V/PS/VS, RG503, RG-Arc) by providing
the center-supply regulator binding and setting status to "okay" in
the shared rgxx3 DTSI.

The center-supply uses vdd_logic (DCDC_REG1, 500-1350mV) which is the
SoC logic voltage rail shared with the DDR memory controller.

Signed-off-by: ROCKNIX <rocknix@rocknix.org>
---
 .../boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi     | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
index XXXXXXX..XXXXXXX 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
@@ -679,3 +679,8 @@ &vp0 {
 		remote-endpoint = <&hdmi_in_vp0>;
 	};
 };
+
+&dmc {
+	center-supply = <&vdd_logic>;
+	status = "okay";
+};
-- 
2.34.1

